Closes #2652 derive_vars_joined_summary: Implement derive_vars_joined_summary()

bundfussr opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

bundfussr commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,396 @@
+#'
+#' @caption Derive cumulative dose before event (`CUMDOSA`)
+#'
+#' @info To the `ADAE` dataset the cumulative actual dose up to the day of the
```
Updated

bundfussr commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,396 @@
+#'   set to the sum of `AVAL`.
+#'   - As `ADY` from `ADEX` is used in `filter_join` (but not in `new_vars`), it
+#'    needs to be specified for `join_vars`.
+#'
```
I've added an item for `join_type`. Is it what you have expected?

manciniedoardo replied:
yes, precisely, thanks

bundfussr commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,396 @@
+#'   missing_values = exprs(CUMDOSE = 0)
+#' )
+#'
+#' @caption Derive weekly score
```
If added something to the title and also a note in the description.

What do you think?

manciniedoardo replied:
looks good, thanks

manciniedoardo commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,396 @@
+#'   "1",        15, "Theft",
+#'   "1",        15, "Fatigue",
+#'   "2",         4, "Parasomnia",
+#'   "3",         2, "Truancy"
```
Interesting that truancy and theft are possible AEs!

manciniedoardo commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,396 @@
+#' @export
+#'
+#' @examplesx
+#'
```
awesome, looks good to me thanks

manciniedoardo commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,409 @@
+#'   Please note that the condition for the number of assessments is specified
+#'   in `new_vars` and not in `filter_join`. This is because the number of
+#'   assessments within the week before the visit should be counted but not the
+#'   number of assessment available for the subject.
```
```suggestion
#'   number of assessments available for the subject.
```

manciniedoardo reviewed (approved):
@bundfussr looks good to me, just suggested a typo fix. awaiting a tech review before merging.

bundfussr commented:
@manciniedoardo , I have added examples for `join_type = "before"` and `first_cond_upper`. Please have a look.

manciniedoardo commented on R/derive_vars_joined_summary.R:
```diff
@@ -0,0 +1,494 @@
+#'
+#' @info The examples focus on the functionality specific to this function. For
+#'   examples of functionality common to all "joined" functions like
+#'   `filter_join`, `join_type`, `first_cond_lower`, ... please see the examples
```
This is not accurate anymore as you have (simple) examples for `join_type` and `first_cond_*`

bundfussr replied:
Good point! Updated.

manciniedoardo reviewed (changes requested):
One minor comment

manciniedoardo commented:
> @manciniedoardo , I have added examples for `join_type = "before"` and `first_cond_upper`. Please have a look.

@rossfarrugia FYI - you may wish to reference the simple examples for `join_type` and `first_cond_*` in the new `derive_vars_joined()` documentation? We have a few examples in various nplaces now but I guess it's about finding a balance to not make any of the pages too long!

rossfarrugia commented:
> @rossfarrugia FYI - you may wish to reference the simple examples for `join_type` and `first_cond_*` in the new `derive_vars_joined()` documentation? We have a few examples in various nplaces now but I guess it's about finding a balance to not make any of the pages too long!

I'd guess it will already get linked in the "See also" section so i personally wouldn't add more links, as i already have examples covering these arguments. I think users will get dizzy if we have them too often having to flick from page to page within the same package. I sometimes feel like that with Wikipedia where it's easy to find yourself in a hole and you forget where you started 😆

manciniedoardo commented:
> > @rossfarrugia FYI - you may wish to reference the simple examples for `join_type` and `first_cond_*` in the new `derive_vars_joined()` documentation? We have a few examples in various nplaces now but I guess it's about finding a balance to not make any of the pages too long!
> 
> I'd guess it will already get linked in the "See also" section so i personally wouldn't add more links, as i already have examples covering these arguments. I think users will get dizzy if we have them too often having to flick from page to page within the same package. I sometimes feel like that with Wikipedia where it's easy to find yourself in a hole and you forget where you started 😆

Makes sense!

bms63 commented on R/derive_vars_joined_summary.R:
Could we not use the mutate(USUBJID = 1) and just put them in the tibble.  Would this also solve the issue of the USUBJID not being the first variable?  It just felt a bit different from most of our other examples. 

![image](https://github.com/user-attachments/assets/898c70e9-9762-4119-8256-fe56fe438e32)

bundfussr replied:
I have updated `planned_visits` to move `USUBJID` to the beginning.

bms63 commented:
@bundfussr Not sure what happened with the templates - they passed in the other PR

I added `EXENDTM` to join_vars and it is working.
 
![image](https://github.com/user-attachments/assets/052f9e03-3855-4922-bbdb-d5f750352d83)

bundfussr commented:
> @bundfussr Not sure what happened with the templates - they passed in the other PR
> 
> I added `EXENDTM` to join_vars and it is working.

I think in the last release we introduced a minor "bug". By mistake, all variables of `dataset_add` were joined. Therefore it didn't matter when you forgot to specify a variable in `join_vars`. This doesn't produce wrong results but increases the memory consumption and reduces the performance. I have restored the previous behavior in this PR (see line 1134 in `derive_joined.R`).

rossfarrugia commented on vignettes/occds.Rmd:
```diff
@@ -277,12 +277,10 @@ adae <- derive_vars_joined(
-  join_vars = exprs(EXSTDTM),
+  join_vars = exprs(EXSTDTM, EXENDTM),
   join_type = "all",
-  order = exprs(EXSTDTM),
   filter_add = (EXDOSE > 0 | (EXDOSE == 0 & grepl("PLACEBO", EXTRT))) & !is.na(EXSTDTM),
-  filter_join = EXSTDTM <= ASTDTM & (ASTDTM <= EXENDTM | is.na(EXENDTM)),
-  mode = "first"
```
FYI @bms63

bundfussr replied:
> would you add a note on this to the vignette above the code extract just to explain this? i think it would be useful for users to consider.

Good idea. Note added.

bms63 commented:
Yay!! new function - so I fixed some merge conflicts but then the man check failed.  we should discuss as next meeting as i guess i did something wrong!?!?!
